### PR TITLE
feat: add optional prometheus metrics endpoint.

### PR DESCRIPTION
**Pull request by pantierra**
## Technical Context
- **Related Issues:** https://github.com/developmentseed/eoAPI/issues/193#issuecomment-4353287562
- **Breaking Changes:** No

## Description

Optional Prometheus metrics via `titiler-pgstac[metrics]`, exposed at `/metrics` when the extra is installed and `TITILER_PGSTAC_API_METRICS_ENABLED=TRUE`.

Uses low-cardinality operation labels (`tiles`, `point`, `search`, `landing`, …) instead of default path-based labels, per the approach discussed in https://github.com/developmentseed/eoAPI/issues/193#issuecomment-4353287562

Custom apps can enable the same instrumentation with `instrument_app(app)` (optional custom scrape path via `endpoint=`).

Concretely, things to discuss if this is desired:

* Without `TITILER_PGSTAC_API_METRICS_ENABLED`, the app still starts and no metrics route is registered (even if `[metrics]` is installed)
* With metrics enabled, `/metrics` returns series labeled by operation (`titiler_pgstac_http_requests_total`, `titiler_pgstac_http_request_duration_seconds`)
* Collection/item/search IDs do not appear in label values
* `/healthz` and the scrape endpoint are excluded and untemplated paths (e.g. bare 404s) are ignored

---

## Checklist
- [x] **Linting:** Code is formatted and linted
- [x] **Tests:** Tests pass. I have included new tests for these changes where applicable.
- [ ] **Edge Cases:** I have manually verified "unhappy paths" and edge cases beyond the basic success criteria (e.g., database connection timeouts, malformed input, strict mapping rejections).
- [x] **Documentation:** I have updated `README.md` to reflect any new environment variables, configuration changes, or breaking updates.
- [x] **Accountability:** I can explain the implementation logic for every line of code submitted.

---

## AI tool usage
 - [x] AI (Copilot or something similar) supported my development of this PR. See our [policy about AI tool use](https://stac-utils.github.io/ai-contribution-policy). Use of AI tools *must* be indicated.
 ---
 > **Policy:** We require a "human-in-the-loop." You are the author and are fully accountable for all submitted code. Please ensure all tool-generated content is thoroughly reviewed before submission to ensure it is not an "extractive contribution" that squanders maintainer time.

### CODE REVIEW

**Comment by vincentsarago** [comment]
Thanks @pantierra 

2 things: 
- we already have `telemetry` provided by titiler factories, how is this different? cc @hrodmn 
- I think we should move the metrics to titiler.core factories if really different from the telemetry

**Comment by pantierra** [comment]
To my understanding factory telemetry is for per-request debugging, so to answer _where time goes inside tile/render paths_. This PR adds Prometheus service level indicators, like request count/latency by low-cardinality `operation` labels for dashboards/alerting. Different signal, export model (OTLP push vs `/metrics` scrape), and cardinality. I see this complementary, not redundant.

Agree we could move a generic HTTP metrics instrument helper to `titiler.core` (app-level, not factory) and keep pgstac-specific operation labels here.

**Comment by vincentsarago** [comment]
more thoughts,

Because metrics are at the app level, there is no real need to move this to titiler-core, because it provides factories to be integrated to apps.

File: titiler/pgstac/metrics.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            should_group_status_codes=True,
+            should_ignore_untemplated=True,
+            excluded_handlers=[r"^/healthz$", rf"^{re.escape(endpoint)}$"],
+        )
+        .add(record_service_metrics)
+        .instrument(app)
+        .expose(app, endpoint=endpoint, include_in_schema=False)
+    )
+
+    _INSTRUMENTED_APPS.add(app)
```
Comment (vincentsarago) [inline titiler/pgstac/metrics.py:87]: @pantierra I'm not sure to fully understand why we need this. How could an app be registered twice?
